### DISCUSSION TITLE
Allow generic Signals

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9916] [General Discussions]
To receive a signal in code the current pseudo-code is:

1. `SENDERNAME.SIGNALNAME.connect(FUNCTION)`

I'd like to be able to do a general signal receive too using just:

2. `SIGNALNAME.connect(FUNCTION)`

For example, this is saying: instead of "look for red signal from watchtower 3" it is just "look for red signal". That way, any generic node can send out that signal, and any node with the right code can receive the signal without knowing/tracking/storing the node name of who sends it out.

This is supported by Godots push for encapsulation and separation.

NOTE: I am proposing keep 1 and add 2 as additional functionality, not replace 1 with 2!, (since sometimes it is useful to have the node named.)

**Comment by shak2** [comment]
So I looked into it and it looks like this functionality is covered by notify_groups() in Scenetree, but what that does exactly is not explained anywhere in docs or the general internet so I have no idea how to use notify_group() or what it does.

Could we please get a page in docs explaining notify_group()?

Maybe it could be added to the Groups tutorial page?

[https://docs.godotengine.org/en/stable/tutorials/scripting/groups.html](https://docs.godotengine.org/en/stable/tutorials/scripting/groups.html)

**Reply by AThousandShips** [reply]
See the documentation for the methods, they work like the respective non-group methods, they're quite self explanatory

**Reply by shak2** [reply]
I found this [https://docs.godotengine.org/en/stable/tutorials/best_practices/godot_notifications.html](https://docs.godotengine.org/en/stable/tutorials/best_practices/godot_notifications.html) which might explain it more. I have not checked it yet.

**Reply by AThousandShips** [reply]
You can just call a method on the group instead, much simpler, just use [`call_group`](https://docs.godotengine.org/en/stable/classes/class_scenetree.html#class-scenetree-method-call-group)
